components/script_traits/lib.rs

```diff
@@ -117,10 +118,25 @@ pub enum LayoutControlMsg {
     PaintMetric(Epoch, u64),
 }
 
+/// The origin where a given load was initiated.
+/// Useful for origin checks, for example before evaluation a JS URL.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub enum LoadOrigin {
+    /// A load originating in the constellation.
+    TopLevel,
+    /// A load originating in webdriver.
+    WebDriver,
+    /// A load originating in script.
+    Script(ImmutableOrigin),
+}
+
 /// can be passed to `LoadUrl` to load a page with GET/POST
 /// parameters or headers
 #[derive(Clone, Debug, Deserialize, Serialize)]
 pub struct LoadData {
+    /// The origin of the page when it started the load.
+    /// Used for same-origin comparisons.
```

Comment:
The comment here disagrees with the comment in the definition of `LoadOrigin`. Is this the origin of the page being loaded, or the origin where the load was initiated?

Reply:
Ok, I've update the comment, and it's supposed to be the origin where the load was initiated.